tests/test_posterior.py

```diff
@@ -0,0 +1,36 @@
+# Test logic in posterior class. In particular when some distributions have default geometry
+
+import pytest
+import numpy as np
+import cuqi
+
+def test_posterior_should_give_Model_priority_for_geometry():
+    A = cuqi.model.Model(lambda x: x.ravel(), range_geometry=10**2, domain_geometry=cuqi.geometry.Image2D((10, 10)))
+    x = cuqi.distribution.Gaussian(0, 1, geometry=10**2) # Default geometry
+    y = cuqi.distribution.Gaussian(A(x), 1)
+    posterior = cuqi.distribution.Posterior(y.to_likelihood(np.zeros(10**2)), x)
+    assert posterior.geometry == A.domain_geometry
+
+def test_posterior_should_give_prior_priority_for_geometry_if_default_model_geometry():
+    A = cuqi.model.Model(lambda x: x.ravel(), range_geometry=10**2, domain_geometry=(10,10))
+    x = cuqi.distribution.Gaussian(0, 1, geometry=cuqi.geometry.Image2D((10, 10)))
+    y = cuqi.distribution.Gaussian(A(x), 1)
+    posterior = cuqi.distribution.Posterior(y.to_likelihood(np.zeros(10**2)), x)
+    assert posterior.geometry == x.geometry
```

Comment:
I looked a bit into this and it seems, `posterior.geometry` is `_DefaultGeometry2D(100,)` while is x.geometry is `Image2D(100,)`. Is this the expected behavior, or is the expected behavior that `posterior.geometry` is `Image2D(100,)`. 

This could be related to how we sat the equivalence logic in geometries (_DefaultGeometry2D is image2D as well).

Reply:
Great catch. Fixed the bug